contrib/annoy-cmake/CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
 option(ENABLE_ANNOY "Enable Annoy index support" ${ENABLE_LIBRARIES})
 
-if (NOT ENABLE_ANNOY)
+if ((NOT ENABLE_ANNOY) OR (SANITIZE STREQUAL "undefined"))
```

Comment:
Intended?

Reply:
I'm testing UBSan locally, so I temporarily disabled it in CI to check all other problems. I still haven't figured out where the error with MSan occurs, because the whole process is similar to other indexes and all the empty checks are done. Do you have any idea?

Reply:
Can you reproduce the msan failure locally?

Your PR which I merged yesterday definitely had the msan problem and potentially more ubsan problems (which were suppressed as the ubsan build was disabled). The uninitialized memory was initially heap-allocated by the call to

```cpp
const auto & column_cut = block.getByName(index_column_name).column->cut(*pos, rows_read);
```
 in MergeTreeIndexAnnoy.cpp. Your fix
```cpp
    if (rows_read == 0)
        return;
```
should do the job. The case that function argument `const Block & block` is unexpectedly empty is handled correctly already.

Guess we should wait for CI results and see if the msan problem is gone.